### PR TITLE
Renders user avatar images in activity feed #4809

### PR DESCRIPTION
**Pull request by sergiogcx**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4809

* Front-end change only:
Live Demo: https://363-sg-avatars--atd-moped-main.netlify.app/moped/projects/273?tab=activity_log

Refactors the CDNAvatar component and it is applied to the activity log.

### CODE REVIEW

File: moped-editor/src/components/CDN/Avatar.js [hunk 1]
```diff
@@ -1,36 +1,72 @@
...
+ * @constructor
+ */
+const CDNAvatar = ({
+  className,
+  src,
+  initials,
+  largeInitials,
+  userColor,
+  useGenericAvatar,
+}) => {
```
Comment (sergiogcx) [inline moped-editor/src/components/CDN/Avatar.js:37]: For starters, here we deconstruct the props and we look for specific fields.

File: moped-editor/src/components/CDN/Avatar.js [hunk 2]
```diff
@@ -1,36 +1,72 @@
...
-      <Typography className={props?.largeInitials ? classes.userInitials : null}>
-        {props?.initials ? props.initials : <Icon>user</Icon>}
-      </Typography>
+      {!!!useGenericAvatar && initials && initials.length > 0 ? (
+        <Typography className={largeInitials ? classes.userInitials : null}>
+          {initials}
+        </Typography>
+      ) : (
+        <Icon>person</Icon>
+      )}
```
Comment (sergiogcx) [inline moped-editor/src/components/CDN/Avatar.js:69]: Basically the behavior is:
1. If there is a source file, use it to get it from the CDN.
2. If there is no src, then check if there is a generic avatar override.
3. If there is no override, then use the initials (if they are provided).
4. If there is an override or there are no initials, then use the Person icon.

File: moped-editor/src/views/projects/projectView/ProjectActivityLog.js [hunk 3]
```diff
@@ -232,13 +232,13 @@ const ProjectActivityLog = () => {
...
+                            <CDNAvatar
                               className={classes.avatarSmall}
-                            >
-                              {getInitials(change?.moped_user)}
-                            </Avatar>
+                              src={change?.moped_user?.picture}
+                              initials={getInitials(change?.moped_user)}
+                              userColor={null}
+                              useGenericAvatar={true}
+                            />
```
Comment (sergiogcx) [inline moped-editor/src/views/projects/projectView/ProjectActivityLog.js:241]: Replace Avatar with our custom component, passing all the necessary fields and overrides.

File: moped-editor/src/components/CDN/Avatar.js [hunk 4]
```diff
@@ -1,36 +1,72 @@
...
-      style={{ backgroundColor: imageSrc ? "white" : user?.userColor }}
+      alt={initials}
+      className={className ? className : classes.avatar}
+      src={imageSource}
+      style={imageStyleOverride}
     >
-      <Typography className={props?.largeInitials ? classes.userInitials : null}>
-        {props?.initials ? props.initials : <Icon>user</Icon>}
-      </Typography>
+      {!!!useGenericAvatar && initials && initials.length > 0 ? (
```
Comment (tillyw) [inline moped-editor/src/components/CDN/Avatar.js:63]: Never seen three "not" operators used in sequence, really interesteing!

Comment (sergiogcx) [reply moped-editor/src/components/CDN/Avatar.js:63]: Basically a "negative truthy" if you will. Essentially we are not assuming useGenericAvatar is provided, so if it isn't we turn the value into a boolean which we then turn into its opposite value using the third "not" operator.

**Review by tillyw — APPROVED** [review]
Works great and the code makes sense to me! Thanks for solving this one, @sergiogcx

**Review by patrickm02L — APPROVED** [review]
Looks good. Ship it!

**Pull request merged**